File: 01gpio/03digital_in/digital_input_wpi.py
import wiringpi as wpi

PIN = 18
wpi.wiringPiSetupGpio()
wpi.pinMode(PIN, wpi.INPUT)
old_val = -1
while True:
    val = wpi.digitalRead(PIN)
    if val != old_val:
        old_val = val
        print(val)

File: 01gpio/03digital_in/digital_input_wpi.js
const wpi = require('node-wiring-pi');
const PIN_BTN = 23;

wpi.wiringPiSetupGpio();
wpi.pinMode(PIN_BTN, wpi.INPUT);

let [val, old] = [-1,-1];
while(true){
	val = wpi.digitalRead(PIN_BTN);
	if(val != old){
		old = val;
		console.log(val);	
	}
}